framework/packages/abstract-client/src/remote_account.rs

```diff
@@ -0,0 +1,398 @@
+        install_configs.push(M::install_config(module_configuration)?);
+
+        self.install_module_remote_internal(install_configs)
+    }
+
+    /// Upgrades the account to the latest version
+    ///
+    /// Migrates manager and proxy contracts to their respective new versions.
+    /// Note that execution will be done through source chain
+    pub fn upgrade(&self, version: ModuleVersion) -> AbstractClientResult<Chain::Response> {
+        let modules = vec![
+            (
+                ModuleInfo::from_id(abstract_std::registry::MANAGER, version.clone())?,
+                Some(
+                    to_json_binary(&abstract_std::manager::MigrateMsg {})
+                        .map_err(Into::<CwOrchError>::into)?,
+                ),
+            ),
+            (
+                ModuleInfo::from_id(abstract_std::registry::PROXY, version)?,
+                Some(
+                    to_json_binary(&abstract_std::proxy::MigrateMsg {})
+                        .map_err(Into::<CwOrchError>::into)?,
+                ),
+            ),
+        ];
+        self.ibc_client_execute(ibc_client::ExecuteMsg::RemoteAction {
+            host_chain: self.host_chain(),
+            action: ibc_host::HostAction::Dispatch {
+                manager_msgs: vec![manager::ExecuteMsg::Upgrade { modules }],
+            },
+        })
+    }
+
+    /// Returns owner of the account
+    pub fn ownership(&self) -> AbstractClientResult<cw_ownable::Ownership<String>> {
+        let manager = self.manager()?;
+        self.remote_chain()
+            .query(&manager::QueryMsg::Ownership {}, &manager)
+            .map_err(Into::into)
+            .map_err(Into::into)
+    }
+
+    /// Returns the owner address of the account.
+    /// If the account is a sub-account, it will return the top-level owner address.
+    pub fn owner(&self) -> AbstractClientResult<Addr> {
+        let mut governance = self
+            .abstr_owner_account
+            .manager
+            .info()?
+            .info
+            .governance_details;
+
+        let environment = self.origin_chain();
+        // Get sub-accounts until we get non-sub-account governance or reach recursion limit
+        for _ in 0..MAX_ADMIN_RECURSION {
+            match &governance {
+                GovernanceDetails::SubAccount { manager, .. } => {
+                    governance = environment
+                        .query::<_, InfoResponse>(&manager::QueryMsg::Info {}, manager)
+                        .map_err(|err| err.into())?
+                        .info
+                        .governance_details;
+                }
+                _ => break,
+            }
+        }
+
+        // Get top level account owner address
+        governance
+            .owner_address()
+            .ok_or(AbstractClientError::RenouncedAccount {})
+    }
+
+    /// Executes a [`CosmosMsg`] on the proxy of the account.
+    /// Note that execution will be done through source chain
+    pub fn execute(
```

Comment:
We need a helper to execute any messages on the manager (not only plain execute cosmosMsg through the proxy). This is also missing on the local ABstract Client. This function is actually rarely used when dealing with modules for instance

Reply:
Nice catch, adding "execute_on_manager" to both accounts!

Reply:
Love it thanks !